File: lib/core/push/foreground_push_notification_policy.dart
import 'package:fluxer_app/core/push/push_notification_payload.dart';

abstract final class ForegroundPushNotificationPolicy {
  ForegroundPushNotificationPolicy._();

  static bool shouldProcessAlertPush({required bool isAppForeground}) =>
      !isAppForeground;

  static bool shouldProcessPush({
    required bool isAppForeground,
    required Map<String, String> payload,
  }) {
    if (isNotificationClearPayload(payload)) {
      return true;
    }
    return shouldProcessAlertPush(isAppForeground: isAppForeground);
  }
}

File: ios/Shared/ForegroundPushNotificationPolicy.swift
import Foundation

enum ForegroundPushNotificationPolicy {
  static func shouldProcessAlertPush(isAppForeground: Bool) -> Bool {
    return !isAppForeground
  }

  static func shouldProcessPush(
    userInfo: [AnyHashable: Any],
    isAppForeground: Bool
  ) -> Bool {
    if PushNotificationPayload.isClearPayload(from: userInfo) {
      return true
    }
    guard PushNotificationPayload.hasApsAlert(userInfo) else {
      return false
    }
    return shouldProcessAlertPush(isAppForeground: isAppForeground)
  }
}
